strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/AllChanges.java

```diff
@@ -0,0 +1,39 @@
+package org.arquillian.smart.testing.vcs.git;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The intention of this class is not to provide a final strategy to final user but to be used by other strategies that requires all changes of Git commits.
+ */
+public class AllChanges {
+
+    private NewFilesDetector newFilesDetector;
+    private ChangedFilesDetector changedFilesDetector;
+
+    public AllChanges(File currentDir, String previous, String head, String ... globPatterns) {
+        newFilesDetector = new NewFilesDetector(currentDir, previous, head, globPatterns);
+        changedFilesDetector = new ChangedFilesDetector(currentDir, previous, head, globPatterns);
+    }
+
+    public Set<String> getTests() {
+        final Set<String> tests = new HashSet<>();
+
+        tests.addAll(newFilesDetector.getTests());
+        tests.addAll(changedFilesDetector.getTests());
```

Comment:
Yes you are right, the problem is that then we get some duplication of code on child classes and moving some classes from parent to child. So for now and since this should be called only once, I decided to do it easy for now (also easy to understand) and when we have this integrated with real projects, then profile it and see if we are loosing time there or not.